### PR TITLE
feat(CX-2923): add recently sold artworks rail to SWA landing page

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description

- Add recently sold artworks rail to SWA landing page

This PR resolves [CX-2923]

https://user-images.githubusercontent.com/11945712/189924735-1fa0e614-bc5a-4da1-bde1-6793f98d44c0.mov

#### Acceptance Criteria

-

#### Setup Instructions

-

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add recently sold artworks rail to SWA landing page - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-2923]: https://artsyproduct.atlassian.net/browse/CX-2923?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/SellWithArtsy/SellWithArtsyHome.tsx [hunk 1]
```diff
@@ -46,7 +56,12 @@ export const SellWithArtsyHome: React.FC<Props> = ({ targetSupply, isLoading })
     <ScrollView>
       <Join separator={<Separator my={3} />}>
         <Header onConsignPress={handleConsignPress} />
-        <RecentlySold targetSupply={targetSupply} isLoading={isLoading} />
+        {enableCustomRecentlySold ? (
+          <SellWithArtsyCustomRecentlySold recentlySoldArtworks={recentlySoldArtworks} />
+        ) : (
+          <RecentlySold targetSupply={targetSupply} isLoading={isLoading} />
+        )}
```
Comment (MrSltun) [inline src/app/Scenes/SellWithArtsy/SellWithArtsyHome.tsx:63]: 👏 👏

File: src/app/Scenes/SellWithArtsy/index.tests.tsx [hunk 2]
```diff
@@ -2,8 +2,9 @@ import { ContextModule, OwnerType } from "@artsy/cohesion"
 import { mockTrackEvent } from "app/tests/globallyMockedStuff"
 import { renderWithWrappersLEGACY } from "app/tests/renderWithWrappers"
 import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils"
+import { __globalStoreTestUtils__ } from "../../store/GlobalStore"
```
Comment (MrSltun) [inline src/app/Scenes/SellWithArtsy/index.tests.tsx:5]: 👀  I see that this is imported but no use for it?

File: src/app/store/config/features.ts [hunk 3]
```diff
@@ -273,6 +273,12 @@ export const features = defineFeatures({
     showInAdminMenu: true,
     echoFlagKey: "AREnableMyCollectionInsightsMedianPrice",
   },
+  ARCustomRecentlySoldOnArtsy: {
+    readyForRelease: true,
+    description: "Use custom recently sold on Artsy",
+    showInAdminMenu: true,
+    echoFlagKey: "ARCustomRecentlySoldOnArtsy",
+  },
```
Comment (MrSltun) [inline src/app/store/config/features.ts:281]: 🙌

**Review by MrSltun — APPROVED** [review]
Great work Mounir! 🚀 🚀 
I just left one comment on the test :)

File: src/app/Scenes/SellWithArtsy/index.tests.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/app/Scenes/SellWithArtsy/index.tests.tsx:5]: wow - how is it possible that ts didn't complain about it, I will remove

**Pull request merged**